### PR TITLE
Change paper model to dataclass

### PR DESCRIPTION
**Pull request by ratuspro**
Changes the Paper Model to a pythons dataclass as suggested on #21.

### CODE REVIEW

**Review by RicardoEPRodrigues — COMMENTED** [review]
I left a question regarding the `PaperIDs` class. I also updated the branch with the changes in `dev`. Please review it.

File: tests/integration/test_semantic_scholar_provider.py [hunk 1]
```diff
@@ -47,23 +47,23 @@ def test_retrieved_paper_externalIds_when_fetching_papers_by_id(self):
...
             PaperIDs.LABEL.DOI), paper_1_externalIds["DOI"])
-        self.assertEqual(paper_1_result.get_external_id(
+        self.assertEqual(paper_1_result.external_ids.get_id(
             PaperIDs.LABEL.CORPUS), paper_1_externalIds["CorpusId"])
 
         paper_2_result = self.__provider.get_paper_by_id(paper_2_id)
         self.assertIsNotNone(paper_2_result)
 
-        self.assertEqual(paper_2_result.get_external_id(
+        self.assertEqual(paper_2_result.external_ids.get_id(
```
[the remark this answers is not in the document]
Comment (hineios) [reply tests/integration/test_semantic_scholar_provider.py:66]: Even if it should, that's outside the scope of this PR.

**Review by hineios — APPROVED** [review]
@ratuspro, you should be more "precise" on you changes.
I don't particularly enjoy the "silent" refactoring of names happening here.

If you are changing a model to use dataclasses you shouldn't be refactoring its slots names.
In my opininion, off course...

If it was the objective all along, be explicit about it! Both in the PR message an in the individual commit messages (you can add more than one line to both).

Nonetheless, your changes and @RicardoEPRodrigues's seem fine.

File: tests/integration/test_semantic_scholar_provider.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hineios) [reply tests/integration/test_semantic_scholar_provider.py:66]: I've created issue #27 to address this question.

**Pull request merged**